lib/core/network/api/aws_amplify/post_image.dart

```diff
@@ -0,0 +1,146 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:amazon_cognito_identity_dart_2/cognito.dart';
+import 'package:amazon_cognito_identity_dart_2/sig_v4.dart';
+import 'package:http/http.dart' as http;
+import 'package:path/path.dart' as path;
+
+// Note: This code based on the ewxample in amazon_cognito_identity_dart_2 flutter package
+class Policy {
+  String expiration;
+  String region;
+  String bucket;
+  String key;
+  String credential;
+  String datetime;
+  String sessionToken;
+  int maxFileSize;
+
+  Policy(
+    this.key,
+    this.bucket,
+    this.datetime,
+    this.expiration,
+    this.credential,
+    this.maxFileSize,
+    this.sessionToken, {
+    this.region = 'us-east-1',
+  });
+
+  factory Policy.fromS3PresignedPost(
+    String key,
+    String bucket,
+    int expiryMinutes,
+    String accessKeyId,
+    int maxFileSize,
+    String sessionToken, {
+    required String region,
+  }) {
+    final datetime = SigV4.generateDatetime();
+    final expiration = DateTime.now().add(Duration(minutes: expiryMinutes)).toUtc().toString().split(' ').join('T');
+    // print('expiration: $expiration');
+    final cred = '$accessKeyId/${SigV4.buildCredentialScope(datetime, region, 's3')}';
+    final p = Policy(key, bucket, datetime, expiration, cred, maxFileSize, sessionToken, region: region);
+    return p;
+  }
+
+  String encode() {
+    final bytes = utf8.encode(toString());
+    return base64.encode(bytes);
+  }
+
+  @override
+  String toString() {
+    // Safe to remove the "acl" line if your bucket has no ACL permissions
+    return '''
+    { "expiration": "$expiration",
+      "conditions": [
+        {"bucket": "$bucket"},
+        ["starts-with", "\$key", "$key"],
+        {"acl": "public-read"},
+        ["content-length-range", 1, $maxFileSize],
+        {"x-amz-credential": "$credential"},
+        {"x-amz-algorithm": "AWS4-HMAC-SHA256"},
+        {"x-amz-date": "$datetime" },
+        {"x-amz-security-token": "$sessionToken" }
+      ]
+    }
+    ''';
+  }
+}
+
+Future<bool> postImage({
+  required CognitoCredentials credentials,
+  required File image,
+  required String fileName,
+  required String s3Region,
+  required String s3Bucket,
+}) async {
+  final _region = s3Region;
+  final String _s3Endpoint = 'https://$s3Bucket.s3.amazonaws.com';
+  final file = image;
+
+  //final stream = http.ByteStream(DelegatingStream.typed(file.openRead()));
+  final stream = http.ByteStream(file.openRead())..cast();
+
+  final length = await file.length();
+
+  final uri = Uri.parse(_s3Endpoint);
+  final req = http.MultipartRequest('POST', uri);
```

Comment:
final NetworkingManager networkingManager;??

Reply:
By way of explanation - I copied this entire code block from the aws amplify example packages. 

I can certainly try making it work with networkingManager - should do. 

Let's make a separate code cleanup ticket for this.

Reply:
https://github.com/hypha-dao/hypha_wallet/issues/88